tortoisebeacon/tortoise_beacon.go

```diff
@@ -460,12 +459,43 @@ func (tb *TortoiseBeacon) proposalPhaseImpl(ctx context.Context, epoch types.Epo
 	}
 
 	logger.With().Info("sent proposal", log.String("message", m.String()))
+	return nil
+}
 
-	tb.mu.Lock()
-	defer tb.mu.Unlock()
-
-	tb.incomingProposals.valid = append(tb.incomingProposals.valid, proposedSignature)
+func (tb *TortoiseBeacon) getProposalChannel(ctx context.Context, epoch types.EpochID) chan *proposalMessageWithReceiptData {
+	tb.mu.RLock()
+	defer tb.mu.RUnlock()
 
+	if epoch < tb.epochInProgress {
```

Comment:
What do you think of rewriting this to a `switch`?

Reply:
done. thanks for the suggestion